Sources/Commands/SwiftTestTool.swift

```diff
@@ -838,19 +866,25 @@ extension SwiftTestTool: ToolName {
 /// Creates the environment needed to test related tools.
 fileprivate func constructTestEnvironment(
     sanitizers: EnabledSanitizers,
-    toolchain: UserToolchain
+    toolchain: UserToolchain,
+    options: ToolOptions,
+    buildParameters: BuildParameters
 ) throws -> [String: String] {
+
     // This only makes sense for macOS.
   #if !os(macOS)
     return Process.env
   #else
+    var env = Process.env
+    if options.shouldEnableCodeCoverage {
+        env["LLVM_PROFILE_FILE"] = buildParameters.buildPath.asString + "/default.profraw"
```

Comment:
We should put the code coverage data in its own directory. Use the `appending` method on `AbsolutePath` to form a new path.

Reply:
done